### PR TITLE
test_tpm2_import: work around spec errata

### PR DESCRIPTION
**Pull request by williamcroberts**
TPM 2.0 spec 1.16 errata on decrypt and sign set for a symcipher.

Certain TPM 2.0 chip, e.g, Nationz Z32H320TC, and tpm2simulator
referring to TPM 2.0 spec 1.16 may have an errata, disallowing both
decrypt and sign set for a symcipher object, and in response to
RC_ATTRIBUTES.

In order to work around it, we attempt to run tpm2_importe again with
sign bit clear.

This change partially resolves the issue #335 (Make sure that all
tests work on a real TPM hardware) for helping on it regard.

Signed-off-by: Jia Zhang <qianyue.zj@alibaba-inc.com>

### CODE REVIEW

File: test/system/test_helpers.sh [hunk 1]
```diff
@@ -114,3 +114,13 @@ create_object() {
         true
     fi
 }
+
+import_object() {
+    local alg_import_obj=0x20040
+
+    if tpm2_import $@ 2>&1 | grep -q 'ERROR: Failed Key Import 000002C2'; then
```
Comment (martinezjavier) [inline test/system/test_helpers.sh]: If this is an errata of 1.16, can we instead of assuming the TPM2_Import is failing due that, check if the TPM2 revision is 1.16 and CLEAR the sign and decrypt bit? This can be checked with something like following:
```
tpm2_dump_capability -c properties-fixed | grep TPM_PT_REVISION | cut -d ':' -f2 | tr -d '[:space:]'
```

Or extend tpm2_dump_capability to only print the revision.

File: tools/tpm2_import.c [hunk 2]
```diff
@@ -564,17 +583,18 @@ static bool on_option(char key, char *value) {
...
-      { "input-key-file",     required_argument, NULL, 'k'},
-      { "parent-key-handle",  required_argument, NULL, 'H'},
-      { "parent-key-public",  required_argument, NULL, 'f'},
-      { "import-key-private", required_argument, NULL, 'r'},
-      { "import-key-public",  required_argument, NULL, 'q'},
+      { "input-key-file",     required_argument, NULL, 'k' },
+      { "parent-key-handle",  required_argument, NULL, 'H' },
+      { "parent-key-public",  required_argument, NULL, 'f' },
+      { "import-key-private", required_argument, NULL, 'r' },
+      { "import-key-public",  required_argument, NULL, 'q' },
```
Comment (martinezjavier) [inline tools/tpm2_import.c]: Please don't change the these entries, if you want to do some style cleanups do it as a separate commit.

File: tools/tpm2_import.c [hunk 3]
```diff
@@ -564,17 +583,18 @@ static bool on_option(char key, char *value) {
...
+      { "object-attributes",  required_argument, NULL, 'A' },
```
Comment (martinezjavier) [inline tools/tpm2_import.c]: This new option has to be added to the man page.

**Comment by williamcroberts** [comment]
FYI The check failing is likely due to tss. Im trouble shooting that now:

File: test/system/test_helpers.sh (hunk 1 above)
Comment (williamcroberts) [reply test/system/test_helpers.sh]: This tool needs a yaml output, that would make this way easier to parse in a programmatic way.

Something that's been building in my head is that each tool outputs a tree of output to tool_main and then tool main can format and output....

**Comment by williamcroberts** [comment]
Ok the bug is in the tss between 3bc96032d47...3f1828ccb31

Ill set up a git bisection and find it.

**Comment by jiazhang0** [comment]
@martinezjavier @williamcroberts Thanks for your comments and review.

I will implement a generic framework to handle errata for tpm2.0-tools these days, implying this patch and previous tpm2_create will be reworked under the new framework. If you agree, plz close this PR. I confirm the new framework would be better than current implementation.

File: tools/tpm2_import.c (hunk 3 above)
Comment (jiazhang0) [reply tools/tpm2_import.c]: The above issues are addressed by https://github.com/01org/tpm2-tools/pull/577.

File: test/system/test_helpers.sh (hunk 1 above)
Comment (jiazhang0) [reply test/system/test_helpers.sh]: I implemented a generic framework (https://github.com/01org/tpm2-tools/pull/578) to handle TPM errata. Plz help to review it.
